evm/src/fuzz/strategies/invariants.rs

```diff
@@ -0,0 +1,171 @@
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+use ethers::{
+    abi::{Abi, Function, ParamType},
+    types::{Address, Bytes},
+};
+use proptest::prelude::*;
+pub use proptest::test_runner::Config as FuzzConfig;
+
+use crate::fuzz::{
+    fuzz_calldata, fuzz_calldata_from_state,
+    invariant::{BasicTxDetails, FuzzRunIdentifiedContracts},
+    strategies::fuzz_param,
+    EvmFuzzState,
+};
+
+/// Given a target address, we generate random calldata.
+pub fn override_call_strat(
+    fuzz_state: EvmFuzzState,
+    contracts: FuzzRunIdentifiedContracts,
+    target: Arc<RwLock<Address>>,
+) -> SBoxedStrategy<(Address, Bytes)> {
+    let contracts_ref = contracts.clone();
+
+    let random_contract = any::<prop::sample::Selector>()
+        .prop_map(move |selector| *selector.select(contracts_ref.read().keys()));
+    let target = any::<prop::sample::Selector>().prop_map(move |_| *target.read());
+
+    proptest::strategy::Union::new_weighted(vec![
+        (80, target.sboxed()),
+        (20, random_contract.sboxed()),
+    ])
+    .prop_flat_map(move |target_address| {
+        let fuzz_state = fuzz_state.clone();
```

Comment:
one of the 2 fuzz_state clones are redundant i think?

Reply:
Unless i'm missing something, it's not. `move` will try to copy `fuzz_state` if you remove any of the clones. Resulting in:

```
move occurs because `fuzz_state` has type `Arc<parking_lot::lock_api::RwLock<parking_lot::RawRwLock, std::collections::HashSet<[u8; 32]>>>`, which does not implement the `Copy` trait
```